### ISSUE TITLE
IDISP007 should warn when pattern matching and disposing.

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#64]
```cs
namespace RoslynSandbox
{
    using System;

    public class Foo
    {
        public static void Bar(object o)
        {
            if (o is IDisposable disposable)
            {
                ↓disposable.Dispose();
            }
        }
    }
}
```
